### PR TITLE
[WIP] Add beats_translate_half_earlier Control

### PR DESCRIPTION
**Pull request by maxjmorley**
Add a skeleton function and control for translating beatgrid half beat earlier.~

### CODE REVIEW

**Comment by JoergAtGithub** [comment]
Welcome at Mixxx!
As a first-time contributor we need you to sign the [Mixxx Contributor Agreement](https://docs.google.com/forms/d/e/1FAIpQLScC9QG327sjLL0eWftmfGUasxFFLxg6LCXJ2xHDYRzFIRqyiw/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ) and comment here when you have done so. It gives us permission to distribute your contribution under the GPL v2 or later license and the Apple Mac App Store. It is also helpful for us to have contact information for contributors in case we may need it in the future.

File: src/controllers/controlpickermenu.cpp [hunk 1]
```diff
@@ -117,7 +117,7 @@ ControlPickerMenu::ControlPickerMenu(QWidget* pParent)
     constexpr int kNumEqRacks = 1;
     const int iNumDecks = static_cast<int>(ControlObject::get(ConfigKey("[Master]", "num_decks")));
     for (int iRackNumber = 0; iRackNumber < kNumEqRacks; ++iRackNumber) {
-        // TODO: Although there is a mode with 4-band EQs, it's not feasible
+        // TODO: Although there is a mode with 4-band EQs, it's not
```
Comment (daschuer) [inline src/controllers/controlpickermenu.cpp:120]: Was this change intended?

File: src/controllers/controlpickermenu.cpp [hunk 2]
```diff
@@ -211,10 +211,14 @@ ControlPickerMenu::ControlPickerMenu(QWidget* pParent)
...
     addDeckAndSamplerControl("beats_translate_later", tr("Move Beatgrid Later"), tr("Adjust the beatgrid to the right"), bpmMenu);
+    addDeckAndSamplerControl("beats_translate_half_earlier", 
+        tr("Move Beatgrid Half Beat Earlier"), 
+        tr("Adjust the beatgrid by a half a beat to the left"),
+        bpmMenu);
     addDeckControl("beats_translate_curpos", tr("Adjust Beatgrid"),
                    tr("Align beatgrid to current position"), bpmMenu);
     addDeckControl("beats_translate_match_alignment", tr("Adjust Beatgrid - Match Alignment"),
-                   tr("Adjust beatgrid to match another playing deck."), bpmMenu);
+                   tr("Adjust beatgrid to match another playing deck."), bpmMenu);
```
Comment (daschuer) [inline src/controllers/controlpickermenu.cpp:221]: extra white space. 
I recommend to install pre-commit that will automatically fix these issues. https://github.com/mixxxdj/mixxx/wiki/Using-Git#set-up-automatic-code-checking

**Review by daschuer — CHANGES_REQUESTED** [review]
Thank you for this PR. Can you tell us how you like to use these new controls?

**Comment by maxjmorley** [comment]
> Thank you for this PR. Can you tell us how you like to use these new controls?

Hey all. I actually made this PR out of habit after my first commit. Just as a WIP pull request. I will go ahead and close it, and reopen one when I actually have a working feature :)
Thanks for the note on pre-commit I will go ahead and set that up now.
